packages/block-library/src/image/edit.js

```diff
@@ -110,10 +110,16 @@ export function ImageEdit( {
 		metadata,
 	} = attributes;
 	const [ temporaryURL, setTemporaryURL ] = useState( attributes.blob );
-	const figureRef = useRef();
 
-	const [ contentResizeListener, { width: containerWidth } ] =
-		useResizeObserver();
+	const containerRef = useRef();
+	// Only observe the max width from the parent container when the parent layout is not flex nor grid.
+	// This won't work for them because the container width changes with the image.
+	// TODO: Find a way to observe the container width for flex and grid layouts.
+	const isMaxWidthContainerWidth =
+		!! parentLayout &&
+		parentLayout.type !== 'flex' &&
+		parentLayout.type !== 'grid';
```

Comment:
In testing a Classic theme, I just discovered that `parentLayout` may not even be present, so the image block can be extended beyond the Column container again:

<img width="1327" alt="image" src="https://github.com/user-attachments/assets/8b6a0397-f19c-4bd1-b38c-f4f7261bf0ff">

So, to handle that, would it work to update this check to something like:

```
	const isMaxWidthContainerWidth =
		! parentLayout || 
		( parentLayout?.type !== 'flex' &&
		parentLayout?.type !== 'grid' );
```

Reply:
Pushed in [9d8fa55](https://github.com/WordPress/gutenberg/pull/64819/commits/9d8fa5561071066b7fafaa7b0a0f35efa5dcfb25)!